### PR TITLE
PIL-1269-WalletConnect In-App Browser

### PR DESCRIPTION
**Pull request by Yash-Chaubal**
Still waiting on the adjusted browser UI design, now that it's a screen instead of a modal!

Features include:
- In-app browser to connect your wallet without leaving the app.
- An address bar so you can go to any URL.
- Back and forward buttons to navigate the browser.
- A refresh button to reload the page.
- A button to stop loading the page.
- A menu button that will open an options modal.
- And a single option that lets you open the dApp in the phone's system browser.

The functionalities are all there, just need to wait to see what the finalised design's like before merging. Thought I'd put this here to get some feedback in the meantime.

https://user-images.githubusercontent.com/55538608/150279439-60a80690-8fff-4f74-93f1-074086a4bc36.mp4

### CODE REVIEW

File: src/components/InAppBrowser/BrowserFloatingActions.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+const ActionsContainer = styled.View`
+  flex-direction: row;
+  justify-content: space-between;
+  height: 64px;
+  background-color: ${({ theme }) => theme.colors.basic090}
+  border-radius: ${borderRadiusSizes.mediumSmall}px;
+  margin-horizontal: ${spacing.extraLarge}px;
+  shadow-opacity: 0.1;
+  shadow-color: #000;
```
Comment (IAmKio) [inline src/components/InAppBrowser/BrowserFloatingActions.tsx:46]: Can we add this to the theme please?

File: src/components/InAppBrowser/InAppBrowser.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  const [urlValue, setUrlValue] = useState(null);
+
+  const [isTyping, setIsTyping] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const [canGoBack, setCanGoBack] = useState(false);
+  const [canGoForward, setCanGoForward] = useState(false);
+
+  useEffect(() => {
+    if (!initialUrl || initialUrl === '') setUrl(DEFAULT_URL);
```
Comment (IAmKio) [inline src/components/InAppBrowser/InAppBrowser.tsx:42]: Could you just check that empty strings evaluate falsely? Then you could just do:

```suggestion
    if (!initialUrl) setUrl(DEFAULT_URL);
```

**Review by IAmKio — CHANGES_REQUESTED** [review]
Some minor points but other than that - really great work!

File: src/components/InAppBrowser/InAppBrowser.tsx (hunk 2 above)
Comment (Yash-Chaubal) [reply src/components/InAppBrowser/InAppBrowser.tsx]: It does indeed evaluate falsely

Comment (IAmKio) [reply src/components/InAppBrowser/InAppBrowser.tsx]: @Yash-Chaubal _that darned javascript!_  😂

**Review by IAmKio — APPROVED** [review]

**Review by divyaBacancy — APPROVED** [review]
Great work 👍

**Pull request merged**